pkg/workflow_execution.go

```diff
@@ -460,6 +459,20 @@ func (c *Client) createWorkflow(namespace string, workflowTemplateID uint64, wor
 		wf.ObjectMeta.Labels = opts.Labels
 	}
 
+	newParameters := make([]wfv1.Parameter, 0)
+	for i := range wf.Spec.Arguments.Parameters {
+		param := wf.Spec.Arguments.Parameters[i]
+		if param.Value != nil {
+			// Strip out whitespace from parameter value
+			formattedValue := strings.Replace(*param.Value, " ", "", -1)
+			*param.Value = strings.Replace(formattedValue, "{{workflow.namespace}}", namespace, -1)
+			*param.Value = strings.Replace(*param.Value, "{{workspace.namespace}}", namespace, -1)
```

Comment:
should this use `formattedValue` too?

Reply:
@rushtehrani Do you mean the 3rd line there?

```
*param.Value = strings.Replace(*param.Value, "{{workspace.namespace}}", namespace, -1)
```

If so, no. It is correct because we replace workflow.namespace above. If we set it to formattedValue we would clear out those changes.

Reply:
Ah got it. Minor: might be cleaner to use `ReplaceAll` - it calls `Replace` with `-1` internally.